panic when timeout

```log
time="2022-12-22T03:40:12+08:00" level=warning msg="group clean call DNSCaller<223.5.5.5:53/udp> failed: read udp 192.168.1.10:44390->223.5.5.5:53: i/o timeout"
time="2022-12-22T03:40:12+08:00" level=info answer=nil cost=2006ms group=_clean q_type=AAAA question=xxx. remote="127.0.0.1:47836"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x758664]

goroutine 86050 [running]:
github.com/wolf-joe/ts-dns/redirector.(*cidrRedirector).Redirect(0xc0001a6000, 0xc00046bdd0, 0x0, 0x5, 0xc00012c1a8)
	/home/worker/git/ts-dns/redirector/redirector.go:93 +0x44
github.com/wolf-joe/ts-dns/redirector.NewRedirector.func1(0x880940, 0xc000121180, 0xc00046bdd0, 0x0, 0x0, 0x0)
	/home/worker/git/ts-dns/redirector/redirector.go:55 +0xca
github.com/wolf-joe/ts-dns/inbound.(*handlerImpl).handle(0xc000128410, 0x8808e8, 0xc0000b8300, 0xc00046bdd0, 0x0)
	/home/worker/git/ts-dns/inbound/handler.go:244 +0x475
github.com/wolf-joe/ts-dns/inbound.(*handlerImpl).ServeDNS(0xc000128410, 0x8808e8, 0xc0000b8300, 0xc00046bdd0)
	/home/worker/git/ts-dns/inbound/handler.go:149 +0x4d
github.com/wolf-joe/ts-dns/inbound.(*handlerWrapper).ServeDNS(0xc000102058, 0x8808e8, 0xc0000b8300, 0xc00046bdd0)
	/home/worker/git/ts-dns/inbound/handler.go:71 +0x4c
github.com/miekg/dns.(*Server).serveDNS(0xc00052e000, 0xc0001ba600, 0x29, 0x200, 0xc0000b8300)
	/home/worker/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:659 +0x2fd
github.com/miekg/dns.(*Server).serveUDPPacket(0xc00052e000, 0xc000673200, 0xc0001ba600, 0x29, 0x200, 0x87e8a8, 0xc0004c4008, 0xc000676ea0, 0x0, 0x0)
	/home/worker/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:603 +0x125
created by github.com/miekg/dns.(*Server).serveUDP
	/home/worker/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:533 +0x395
```

--- a/inbound/handler.go
+++ b/inbound/handler.go
@@ -241,7 +241,7 @@ func (h *handlerImpl) handle(writer dns.ResponseWriter, req *dns.Msg) (resp *dns
 	_info.matched = matched
 
 	// redirect
-	if resp != nil && h.redirector != nil {
+	if h.redirector != nil {
 		if group := h.redirector(matched, req, resp); group != nil {
 			matched = group
 			resp = group.Handle(req)

--- a/inbound/handler.go
+++ b/inbound/handler.go
@@ -3,6 +3,12 @@ package inbound
 import (
 	"errors"
 	"fmt"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
 	"github.com/miekg/dns"
 	"github.com/sirupsen/logrus"
 	"github.com/wolf-joe/ts-dns/cache"
@@ -10,11 +16,6 @@ import (
 	"github.com/wolf-joe/ts-dns/hosts"
 	"github.com/wolf-joe/ts-dns/outbound"
 	"github.com/wolf-joe/ts-dns/redirector"
-	"strconv"
-	"strings"
-	"sync/atomic"
-	"time"
-	"unsafe"
 )
 
 // region interface
@@ -240,7 +241,7 @@ func (h *handlerImpl) handle(writer dns.ResponseWriter, req *dns.Msg) (resp *dns
 	_info.matched = matched
 
 	// redirect
-	if h.redirector != nil {
+	if resp != nil && h.redirector != nil {
 		if group := h.redirector(matched, req, resp); group != nil {
 			matched = group
 			resp = group.Handle(req)

--- a/redirector/redirector.go
+++ b/redirector/redirector.go
@@ -3,14 +3,15 @@ package redirector
 import (
 	"bufio"
 	"fmt"
+	"net"
+	"os"
+	"strings"
+
 	"github.com/miekg/dns"
 	"github.com/sirupsen/logrus"
 	"github.com/wolf-joe/ts-dns/config"
 	"github.com/wolf-joe/ts-dns/outbound"
 	"github.com/yl2chen/cidranger"
-	"net"
-	"os"
-	"strings"
 )
 
 const (
@@ -49,7 +50,7 @@ func NewRedirector(globalConf config.Conf, groups map[string]outbound.IGroup) (R
 	// return runtime redirector
 	redirector := func(src outbound.IGroup, req, resp *dns.Msg) outbound.IGroup {
 		instance, exists := group2redir[src.Name()]
-		if !exists {
+		if resp == nil || !exists {
 			return nil
 		}
 		newGroup := instance.Redirect(req, resp)

